source/application/SXRMB/SXRMBAppController.cpp

```diff
@@ -459,18 +407,14 @@ void SXRMBAppController::onCurrentScanActionFinishedImplementation(AMScanAction
 void SXRMBAppController::onUserConfigurationLoadedFromDb()
 {
 	AMXRFDetector *detector = SXRMBBeamline::sxrmb()->brukerDetector();
-
-	foreach (AMRegionOfInterest *region, userConfiguration_->regionsOfInterest()){
-		detector->addRegionOfInterest(region->createCopy());
-		microProbe2DScanConfiguration_->addRegionOfInterest(region);
-		exafsScanConfiguration_->addRegionOfInterest(region);
-		microProbe2DOxidationScanConfiguration_->addRegionOfInterest(region);
-	}
-
 	// This is connected here because we want to listen to the detectors for updates, but don't want to double add regions on startup.
 	connect(detector, SIGNAL(addedRegionOfInterest(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestAdded(AMRegionOfInterest*)));
 	connect(detector, SIGNAL(removedRegionOfInterest(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestRemoved(AMRegionOfInterest*)));
 	connect(detector, SIGNAL(regionOfInterestBoundingRangeChanged(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestBoundingRangeChanged(AMRegionOfInterest*)));
+
+	foreach (AMRegionOfInterest *region, userConfiguration_->regionsOfInterest()){
```

Comment:
This is the slot for signal `loadedFromDb()`. When we load the user configuration from the db in function `setupUserConfiguration()`, it will emit the signal. 

Therefore, the ROIs of the user configuration are created when we execute  `userConfiguration_->loadFromDb(AMDatabase::database("user"), 1)`.